CRM/MembershipExtras/DTO/ScheduleInstalmentAmount.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+
+class CRM_MembershipExtras_DTO_ScheduleInstalmentAmount {
```

Comment:
@omarabuhussein  they are actually different objects.

Reply:
> we would need to use array for storing instalment date and instalment amount inside the Schedule instalment class

Can you explain a bit what do you mean by this, better with little code example ? because I don't understand why you would need to use an array inside ScheduleInstalment class if you are going to decompose these two DTOs.

Reply:
@omarabuhussein  what I meant, was not actually to have an array inside ScheduleInstalment but we would need to use the array for instalment amount, so we might store instalment amount as an array too. 

So why I don't want to decompose these DTOs.  The ScheduleInstalmentAmount class is used as an object for storing the instalment amount that we calculate and it will be passed to calucateFirstInstalment method for getting first instalment amount (depending on membership type / period). See below

https://github.com/compucorp/uk.co.compucorp.membershipextras/blob/MAE-380-membership-type-rolling-period-instalment/CRM/MembershipExtras/Service/MembershipInstalmentsSchedule.php#L158

https://github.com/compucorp/uk.co.compucorp.membershipextras/blob/MAE-380-membership-type-rolling-period-instalment/CRM/MembershipExtras/Service/MembershipInstalmentsSchedule.php#L175

So the actual schedule instalment https://github.com/compucorp/uk.co.compucorp.membershipextras/blob/MAE-380-membership-type-rolling-period-instalment/CRM/MembershipExtras/Service/MembershipInstalmentsSchedule.php#L100

So if we are to decompose it, we might need to change or rename below function from calculateInstalmentAmount() to something like calculateInstalment() or getInstalment() instead?

https://github.com/compucorp/uk.co.compucorp.membershipextras/blob/MAE-380-membership-type-rolling-period-instalment/CRM/MembershipExtras/Service/MembershipInstalmentsSchedule.php#L136

But at the end, we would need ScheduleInstalment DTO would need to re-initiate when we generate an array that contain a list of different instalment objects here. https://github.com/compucorp/uk.co.compucorp.membershipextras/blob/MAE-380-membership-type-rolling-period-instalment/CRM/MembershipExtras/Service/MembershipInstalmentsSchedule.php#L100